spark/v4.0/spark/src/test/java/org/apache/iceberg/spark/source/TestStructuredStreaming.java

```diff
@@ -302,4 +303,12 @@ private <T> MemoryStream<T> newMemoryStream(int id, SQLContext sqlContext, Encod
   private <T> void send(List<T> records, MemoryStream<T> stream) {
     stream.addData(JavaConverters.asScalaBuffer(records));
   }
+
+  private void deleteFileAndCrc(File file) throws IOException {
+    File crcFile = new File(file.getParent(), "." + file.getName() + ".crc");
```

Comment:
Are you sure the "." +file.getName is correct here?

Could we use 
```new File(file.getAbsolutePath + ".crc")```

Also under what cases should crcFile not exist?

Reply:
file.getAbsolutePath + ".crc" has
```
/var/folders/lp/nhx1yjj90qb54xq7v_grpl9c0000gn/T/junit3203001525697015981/parquet/checkpoint/commits/1.crc
```
but I need 
```
/var/folders/lp/nhx1yjj90qb54xq7v_grpl9c0000gn/T/junit3203001525697015981/parquet/checkpoint/commits/.1.crc
```

When deleting the commits like 
```
      // remove the last commit to force Spark to reprocess batch #1
      File lastCommitFile = new File(checkpoint.toString() + "/commits/1");
      assertThat(lastCommitFile.delete()).as("The commit file must be deleted").isTrue();
```
I got Exception so I manually delete the crc file
```
 Caused by:
            org.apache.hadoop.fs.FileAlreadyExistsException: Rename destination file:/var/folders/lp/nhx1yjj90qb54xq7v_grpl9c0000gn/T/junit16535101756839742517/parquet/checkpoint/commits/.1.crc already exists.
                at app//org.apache.hadoop.fs.AbstractFileSystem.renameInternal(AbstractFileSystem.java:876)
                at app//org.apache.hadoop.fs.AbstractFileSystem.rename(AbstractFileSystem.java:807)
                at app//org.apache.hadoop.fs.ChecksumFs.renameInternal(ChecksumFs.java:519)
                at app//org.apache.hadoop.fs.AbstractFileSystem.rename(AbstractFileSystem.java:807)
                at app//org.apache.hadoop.fs.FileContext.rename(FileContext.java:1044)
                at app//org.apache.spark.sql.execution.streaming.FileContextBasedCheckpointFileManager.renameTempFile(CheckpointFileManager.scala:376)
                at app//org.apache.spark.sql.execution.streaming.CheckpointFileManager$RenameBasedFSDataOutputStream.close(CheckpointFileManager.scala:156)
                at app//org.apache.spark.sql.execution.streaming.HDFSMetadataLog.write(HDFSMetadataLog.scala:207)
```